internal/fs/implicit_dirs_test.go

```diff
@@ -569,3 +574,179 @@ func (t *ImplicitDirsTest) AtimeCtimeAndMtime() {
 	ExpectThat(ctime, timeutil.TimeNear(mountTime, delta))
 	ExpectThat(mtime, timeutil.TimeNear(mountTime, delta))
 }
+
+// Create objects in implicit directories with
+// unsupported object names and
+// test that stat and ReadDirPicky on the different directories.
+func (t *ImplicitDirsTest) UnsupportedDirNames() {
+	var fi os.FileInfo
+	var entries []os.FileInfo
+	var err error
+
+	// Set up contents.
+	AssertEq(
+		nil,
+		t.createObjects(
+			map[string]string{
+				"foo//bar": "", // unsupported
+				"foo/1":    "", // supported
+				"a/2":      "", // supported
+				"a//2/6":   "", //unsupported
+				"a//3":     "", // unsupported
+				"4":        "", // supported
+				"/4/7":     "", //unsupported
+				"/bar":     "", // unsupported
+				"bar//5":   "", // unsupported
+				"/":        "", //unsupported
+			}))
+
+	// Statting the mount directory should return a directory entry.
```

Comment:
Should we use strecher testify library?

Reply:
that's a separate effort ongoing in https://github.com/GoogleCloudPlatform/gcsfuse/pull/2006 .